runtime/src/proxy.rs

```diff
@@ -0,0 +1,120 @@
+                                                            // RuntimeCall::Indices(pallet_indices::Call::freeze{..}) |
+                                                            // RuntimeCall::Registrar(paras_registrar::Call::register {..}) |
+                                                            // RuntimeCall::Registrar(paras_registrar::Call::deregister {..}) |
+                                                            // // Specifically omitting Registrar `swap`
+                                                            // RuntimeCall::Registrar(paras_registrar::Call::reserve {..}) |
+                                                            // RuntimeCall::Crowdloan(..) |
+                                                            // RuntimeCall::Slots(..) |
+                                                            // RuntimeCall::Auctions(..) | // Specifically omitting the entire XCM Pallet
+                                                            // RuntimeCall::NominationPools(..) |
+                                                            // RuntimeCall::FastUnstake(..) |
+                                                            // RuntimeCall::Claims(..)
+
+                // zkVerify specifics
+                RuntimeCall::Poe(..) |
+                RuntimeCall::SettlementFFlonkPallet(..) |
+                RuntimeCall::SettlementZksyncPallet(..) |
+                RuntimeCall::SettlementGroth16Pallet(..) |
+                RuntimeCall::SettlementRisc0Pallet(..) |
+                RuntimeCall::SettlementUltraplonkPallet(..)
+            ),
+            ProxyType::Governance => matches!(
+                c,
+                RuntimeCall::Treasury(..)
+                    | RuntimeCall::Bounties(..)
+                    | RuntimeCall::Utility(..)
+                    | RuntimeCall::ChildBounties(..)
+                    | RuntimeCall::ConvictionVoting(..)
+                    | RuntimeCall::Referenda(..)
+                    | RuntimeCall::Whitelist(..)
+            ),
+            ProxyType::Staking => {
+                matches!(
+                    c,
+                    RuntimeCall::Staking(..)
+                        | RuntimeCall::Session(..)
+                        | RuntimeCall::Utility(..)
+                        | RuntimeCall::VoterList(..) // RuntimeCall::FastUnstake(..) |
+                                                     // RuntimeCall::NominationPools(..)
+                )
+            }
+            ProxyType::CancelProxy => {
+                matches!(
+                    c,
+                    RuntimeCall::Proxy(pallet_proxy::Call::reject_announcement { .. })
+                )
+            }
+        }
+    }
+    fn is_superset(&self, o: &Self) -> bool {
+        match (self, o) {
+            (x, y) if x == y => true,
+            (ProxyType::Any, _) => true,
+            (_, ProxyType::Any) => false,
+            (ProxyType::NonTransfer, _) => true,
```

Comment:
That seams the most tricky rule because it claims that all no `Any` poxies should be a sub set of `NonTransfer`.

It should be nice if we can find a way to test it but I cannot see how to find a simple way to create all possible transactions in the other proxies.

Reply:
We might work around the problem expanding line 116 into explicit cases (e.g. `(ProxyType::NonTransfer, ProxyType::Governance) => true`, ...).
That would make this safer against future additions of ProxyTypes at the cost of 2 additional lines, but honestly I don't like this option much.

Even simpler, we might just add a comment between lines `21-22` saying not to add any values in that position :)